src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/NestedFieldProjection.java

```diff
@@ -41,10 +43,93 @@
+ *  |                   |          "path": "projects"                                    |
+ *  |                   |          }                                                     |
+ *  |                   |        },                                                      |
+ *  |                   |        {                                                       |
+ *  |                   |          "nested": {                                           |
+ *  |                   |            "query":{},                                         |
+ *  |                   |            "path": "comments.message"                          |
+ *  |                   |          }                                                     |
+ *  |                   |        }                                                       |
+ *  |                   |      ]                                                         |
+ *  |------------------------------------------------------------------------------------|
+ *  | comments.message  |      "must": [                                                 |
+ *  |                   |        {                                                       |
+ *  |                   |          "nested": {                                           |
+ *  |                   |            "query": {                                          |
+ *  |                   |              "nested": {                                       |
+ *  |                   |                "query":{},                                     |
+ *  |                   |                "path": "projects.address"                      |
+ *  |                   |               }                                                |
+ *  |                   |             },                                                 |
+ *  |                   |          "path": "projects"                                    |
+ *  |                   |          }                                                     |
+ *  |                   |        },                                                      |
+ *  |                   |        {                                                       |
+ *  |                   |          "nested": {                                           |
+ *  |                   |            "query": {                                          |
+ *  |                   |              "nested": {                                       |
+ *  |                   |                "query":{},                                     |
+ *  |                   |                "path": "comments.message"                      |
+ *  |                   |               }                                                |
+ *  |                   |             },                                                 |
+ *  |                   |          "path": "comments"                                    |
```

Comment:
does the comments be added only after processing comments?

Reply:
will add another column for `cur path`